metrics/ci_integrations/docs/diagrams/ci_integrations_config_validator_destination_class_diagram.puml

```diff
@@ -1,54 +1,64 @@
-            abstract class ConfigValidator<T extends Config, V extends ValidationDelegate, R extends ValidationResult> {
-                + validationDelegate : V
-                + validationResultBuilder : ValidationResultBuilder<R>
+    package integration {
+        package interface {
+            package base.config {
+                package model as interface_model {
+                    abstract class ConfigField {}
+                }
+                
+                package validator as base_validator {
+                    interface ConfigValidator<T extends Config> {
+                        + validationDelegate : ValidationDegate
 
-                + Future<R> validate(T config)
+                        + Future<ValidationResult> validate(T config)
+                    }
+                }
             }
-        }
 
-        package destination.config as interface_config {
+            package destination.config as interface_config {
+                package validation_delegate as interface_delegate {
+                    interface DestinationValidationDelegate {
+                        + Future<InteractionResult> validateAuth(AuthorizationBase auth)
+                        + Future<InteractionResult> validateDestinationProjectId(String destinationProjectId)
+                    }
+                }
 
-            package validation_delegate as interface_delegate {
-                interface DestinationValidationDelegate {
-                    + Future<InteractionResult> validateAuth(AuthorizationBase auth)
-                    + Future<InteractionResult> validateDestinationProjectId(String destinationProjectId)
+                package validator_factory as interface_factory {
+                    interface ConfigValidatorFactory<T extends Config> {
+                        + ConfigValidator<T> create(T config)
+                    }
                 }
             }
+        }
 
-            package validator_factory as interface_factory {
-                interface ConfigValidatorFactory<V extends ConfigValidator> {
-                    + V create()
-                }
+        package validation.model as validation_model {
+            package builder as validation_model_builder {
+                class ValidationResultBuilder {}
             }
 
-            package model as interface_model {
-                package builder {
-                    abstract class DestinationValidationResultBuilder {
-                        + authValidationResult : FieldValidationResult
-                        + destinationProjectIdValidationResult : FieldValidationResult
+            class ValidationResult {}
 
-                        + void setAuthValidationResult(FieldValidationResult authValidationResult)
-                        + void setDestinationProjectIdValidationResult(FieldValidationResult projectIdValidationResult)
-                    }
-                }
+            class FieldValidationResult {
```

Comment:
Would it be better to display this class on the interfaces diagram? The same suggestion for the `FieldValidationConclusion` enum.